frontends/p4/validateParsedProgram.cpp

```diff
@@ -175,11 +175,21 @@ void ValidateParsedProgram::postorder(const IR::EntriesList* l) {
         ::error("%1%: table initializers must be constant", l);
 }
 
-/// Switch statements are not allowed in actions
+/// Switch statements are not allowed in actions.
+/// Default label in switch statement is always the last one.
 void ValidateParsedProgram::postorder(const IR::SwitchStatement* statement) {
     auto inAction = findContext<IR::P4Action>();
     if (inAction != nullptr)
         ::error("%1%: switch statements not allowed in actions", statement);
+    bool defaultFound = false;
+    for (auto c : statement->cases) {
+        if (defaultFound) {
+            ::error("%1%: label following default label", c);
+            break;
+        }
+        if (c->label->is<IR::DefaultExpression>())
+            defaultFound = true;
+    }
```

Comment:
How about the following:
```
    bool defaultFound = false;
    std::vector<IR::SwitchCase *> labelsAfterDefault;
    for (auto c : statement->cases) {
        if (defaultFound)
            labelsAfterDefault.push_back(c);
        if (c->label->is<IR::DefaultExpression>())
            defaultFound = true;
    }
   // since this may break existing programs, we should throw an error only if in strict compilation mode
   // otherwise, print a warning and let folks know that it may become an error
    if (options.isStrict())
         ::error("%1%: Switch statements require the default label to be the last one. The following labels are listed following the default: %2%", statement, labelsAfterDefault);
    else
         ::warning("%1%: Switch statements will require the default label to be the last one. This behavior will be enforced in a future version of the compiler. The following labels are listed following the default: %2%", statement, labelsAfterDefault);
```

It will of course, require adding the strict option, but I do believe that is better than breaking existing programs outside the p4lang/p4c test suite.

Reply:
We can just make this a warning.
We have -Werror to be strict.

Reply:
That's a good idea. And we then we can turn it into an error later when people had a chance to migrate their code.